src/deploy/functions/tasks.ts

```diff
@@ -110,122 +124,92 @@ export function updateFunctionTask(
         clc.bold(helper.getFunctionLabel(fn.name)) +
         "..."
     );
-    params.timer.startTimer(fn.name, "update");
     const eventType = fn.eventTrigger ? fn.eventTrigger.eventType : "https";
-    try {
-      const updateRes = await cloudfunctions.updateFunction({
-        projectId: params.projectId,
-        region: helper.getRegion(fn.name),
-        eventType: eventType,
-        functionName: helper.getFunctionId(fn.name),
-        entryPoint: fn.entryPoint,
-        trigger: helper.getFunctionTrigger(fn),
-        labels: Object.assign({}, deploymentTool.labels(), fn.labels),
-        sourceUploadUrl: params.sourceUrl,
-        sourceToken: params.sourceToken,
-        runtime: params.runtime,
-        availableMemoryMb: fn.availableMemoryMb,
-        timeout: fn.timeout,
-        maxInstances: fn.maxInstances,
-        environmentVariables: fn.environmentVariables,
-        vpcConnector: fn.vpcConnector,
-        vpcConnectorEgressSettings: fn.vpcConnectorEgressSettings,
-        serviceAccountEmail: fn.serviceAccountEmail,
-        ingressSettings: fn.ingressSettings,
-      });
-      const pollerOptions: OperationPollerOptions = Object.assign(
-        {
-          pollerName: `update-${fn.name}`,
-          operationResourceName: updateRes.name,
-          onPoll,
-        },
-        defaultPollerOptions
-      );
-      const operationResult = await pollOperation<CloudFunctionTrigger>(pollerOptions);
-      params.timer.endTimer(fn.name);
-      helper.printSuccess(fn.name, "update");
-      return operationResult;
-    } catch (err) {
-      params.errorHandler.record("error", fn.name, "update", err.message || "");
-    }
+    const updateRes = await cloudfunctions.updateFunction({
+      projectId: params.projectId,
+      region: helper.getRegion(fn.name),
+      eventType: eventType,
+      functionName: helper.getFunctionId(fn.name),
+      entryPoint: fn.entryPoint,
+      trigger: helper.getFunctionTrigger(fn),
+      labels: Object.assign({}, deploymentTool.labels(), fn.labels),
+      sourceUploadUrl: params.sourceUrl,
+      sourceToken: sourceToken,
```

Comment:
So where does this come from now if not the params above?

Reply:
Same place as before (the onPollFn in runRegionalDeployment), I just moved it out of the TaskParams object and passed it into this function separately. 

I found a bug while developing where, because every task was sharing the same TaskParam object, we'd overwrite the sourceToken, and cause some deployments to fail because we sent a sourceToken from a different region.

I could have fixed it by making a new TaskParams for every region, but I figured it was cleaner and less bug prone to separate it - having a mutating object passed around and shared by a bunch of simultaneously running async functions seemed poor.